js/app/cookies-banner.js

```diff
@@ -1,24 +1,39 @@
+// get preferences cookie
+var cookiesSet = hasCookiesPreferencesSet()
+
 // cookies settings
 var oneYearInSeconds = 31622400;
 var url = window.location.hostname;
 var cookiesDomain = extractDomainFromUrl(url);
 var cookiesPreference = true;
 var encodedCookiesPolicy = "%7B%22essential%22%3Atrue%2C%22usage%22%3Atrue%7D";
 var cookiesPath = "/";
+var cookiesBanner = $(".js-cookies-banner-form");
+
+document.addEventListener("DOMContentLoaded", determineWhetherToRenderBanner())
+
+function determineWhetherToRenderBanner() {
+    cookiesBanner.addClass("hidden");
+
+    if (!cookiesSet) {
```

Comment:
I am curious is this actually preventing the screen from jittering?
1. Page is loading
2. Script loads in
3.  var cookiesSet = hasCookiesPreferencesSet()  is called
4. banner hidden
5. instantly banner is shown
6. banner is initialised.

If this is the solution then I would suggest:
- Move the `var cookiesSet = hasCookiesPreferencesSet()` to after the element is hidden.
- Call `initCookiesBanner` before making it visible (before  `cookiesBanner.removeClass("hidden");`)

However as stated, I am not sure if this is preventing any jitter (maybe jitter is just not an issue).
Worth trying just:
```
if (cookiesSet) {
    cookiesBanner.addClass("hidden");
} 
else {
    initCookiesBanner();
}
```

Reply:
So I had a look into this and I agree that the logic I had was broken down more than needed. To that end I've simplified as per your suggestion at the bottom. I don't see any jitter when navigating between pages, but it appears that there still is some when navigating to the site via the address bar or hard refresh. I think this remaining jitter needs to be addressed with CSS rather than JS, but I'm not sure what the best approach will be that bears in mind the non-js journey. Will continue to investigate